proto/network.proto

```diff
@@ -13,9 +13,11 @@ message HTTPRequestMsg {
     string url = 2;
     string method = 3;
     bytes body = 4;
+    map <string, string> headers = 5;
 }
 
 message HTTPResponseMsg {
     uint32 responseCode = 1;
     bytes data = 2;
+    bytes headers = 3;
```

Comment:
Can we discuss this? I’m really uncomfortable passing the Response Header as a JSON object serialized to `bytes` instead of a `map <string,string>` like request headers. At some point they will need to be deserialized to key-value pairs before they're exposed to LDK consumers. At the very least, I'd like to understand the considerations that drove this decision.

Reply:
I did it this way because I couldn't find a way to convert the resulting headers from the `http.Header` type to `map[string][]string` so it can be passed through the proto, probably because I'm dumb.

Reply:
Let me tinker with it a little bit more to see if I can get it looking better

Reply:
I was hoping to not have to do something like this to achieve this.
`	responseHeaders := make(map[string][]string)

	for name, value := range resp.Header {
		responseHeaders[name] = value
	}`

Reply:
I'm open to suggestions

Reply:
TBH I think there'll be a bit of tedium here to convert the headers from the Go map of string arrays that's provided for in the `net/http/Header` type to a proto `map<string, string>`. This is the sort of thing that would be easy to represent if Go supported generics, but since it doesn't we gotta roll these mapping functions ourselves. So it's what you said above, but also converting the value from `[]string` to `string` (seperated by commas)